### ISSUE TITLE
Package request: rtl-sdr-blog

### ISSUE THREAD

**Issue by nonnull-ca** [NixOS/nixpkgs#279621]
**Project description**

[rtl-sdr-blog](https://github.com/rtlsdrblog/rtl-sdr-blog) is a fork of [rtl-sdr](https://github.com/osmocom/rtl-sdr) with support for the RTL-SDR-BLOG dongles. The upstream rtl-sdr does not support the RTL-SDR-BLOG-V4. rtl-sdr is [already](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/radio/rtl-sdr/default.nix) packaged.

I'm not sure of the correct way to handle this. Upstream and said fork are mostly compatible for downstream users, but have mildly diverged (+79/-10 commits). For now I have been doing a package override:

```
nixpkgs.config.packageOverrides = pkgs: {
  # Replace rtl-sdr with rtl-sdr-blog, because the vanilla version does not appear to work with the rtl-sdr-blog dongle.
  rtl-sdr = pkgs.rtl-sdr.overrideAttrs (oldAttrs: {
    src = pkgs.fetchFromGitHub {
      owner = "rtlsdrblog";
      repo = "rtl-sdr-blog";
      rev = "V1.3.4";
      hash = "sha256-4Qeawd8V68j9Gfw+AgLXHhiiTpyJM1/eb16AZCTa0Vc=";
    };
  });
};
```

...which 'works', but which forces me to rebuild every downstream package. (And requires manual updates.) (And prevents me from using the vanilla rtl-sdr library without additional work although that is not an issue for me currently.)

**Metadata**

* homepage URL: https://www.rtl-sdr.com/rtl-sdr-quick-start-guide/
* source URL: https://github.com/rtlsdrblog/rtl-sdr-blog
* license: gpl2
* platforms: linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by tw0flower** [comment]
I have installed both and while the rtl-sdr-osmocom did allow me to receive signal from the dongle, it was overly saturated. Even with a 0 db gain the waterfall in SDR++ was bright yellow.

On the other hand, with the packageOverride @nonnull-ca describes, I was able to use the dongle. 

Did you do anything specific to make it work @crazyminecuber ?

I also think it would be nice to have the rtl-sdr-blog modified library available as a binary. While most CLI applications and SDR++ are fairly okay to build I had to give up on building SDRAngel on an older laptop as it would have taken hours for the build to finish.

I couldn't find any discussions about integrating the specific modifications made by the rtl-sdr-blog team to the upstream osmocom version and so I'm doubtful it will happen in the near future.

**Comment by crazyminecuber** [comment]
I did this to overlay the osmocom driver since gqrx for example is built against it.
 ```
nixpkgs.overlays = [
    (final: prev: {
      rtl-sdr = prev.rtl-sdr-osmocom;
    })
  ];
```

But I have not tested that it works 100 % perfectly but at least it seemed to be working and I did not find any obvious issues with oversaturation. How did you "install" the driver?

What specific modifications are you talking about that the blog driver has done exactly? This and a few later commits seem to add support for the v4 dongle in the osmocom driver [https://gitea.osmocom.org/sdr/rtl-sdr/commit/1261fbb285297da08f4620b18871b6d6d9ec2a7b](https://gitea.osmocom.org/sdr/rtl-sdr/commit/1261fbb285297da08f4620b18871b6d6d9ec2a7b)
